selftests/unit/test_utils_params.py

```diff
@@ -86,6 +86,96 @@ def testGetItemMissing(self):
 
+    def testCfgEnabled(self):
+        self.params["something"] = "no"
+        self.params["foobar"] = "yes"
+
+        self.assertEqual(False, params.is_cfg_enabled('something'))
+        self.assertEqual(False, params.is_cfg_enabled('notgiven'))
+        self.assertEqual(True, params.is_cfg_enabled('foobar'))
+
+    def testMultiParamNone(self):
+        self.assertEqual(0, len(self.params.get_multi_param("something")))
+
+    def testMultiParamOnlyOne(self):
+        self.params['something'] = 'foobar'
+
+        found = self.params.get_multi_param("something")
+        self.assertEqual(1, len(found))
+        self.assertEqual('foobar', found[0])
+
+    def testMultiParamMany(self):
+        self.params['something'] = 'foobar'
+        self.params['something2'] = 'foobar2'
+        self.params['something3'] = 'foobar3'
+
+        found = self.params.get_multi_param("something")
+        self.assertEqual(3, len(found))
+        self.assertEqual('foobar', found[0])
+        self.assertEqual('foobar2', found[1])
+        self.assertEqual('foobar3', found[2])
+
+    def testIsObjectSpecific(self):
+        self.assertTrue(is_object_specific("genie_vm1", ["vm1", "vm2"]))
+        self.assertFalse(is_object_specific("genie_vm1", ["nic1", "nic2"]))
+        self.assertTrue(is_object_specific("god_vm2", ["vm1", "vm2"]))
+        self.assertFalse(is_object_specific("god_vm2", ["nic1", "nic2"]))
+        self.assertFalse(is_object_specific("genie", ["vm1", "vm2"]))
+        self.assertFalse(is_object_specific("god", ["vm1", "vm2"]))
+        self.assertFalse(is_object_specific("wizard", ["vm1", "vm2"]))
+        self.assertFalse(is_object_specific("wizard", ["wizard"]))
+
+    def testObjectParams(self):
+        vm_params = utils_params.Params({"name_vm1": "josh", "name_vm2": "jean", "name": "jarjar", "surname": "jura"})
+        params = object_params(vm_params, "vm1", ["vm1", "vm2"])
+        for key in params.keys():
+            self.assertFalse(is_object_specific(key, ["vm1", "vm2"]))
+
+    def testObjectifyParams(self):
+        params = utils_params.Params({"name_vm1": "josh", "name_vm2": "jean", "name": "jarjar", "surname": "jura"})
+        vm_params = objectify_params(params, "vm1", ["vm1", "vm2"])
+        self.assertTrue("name_vm1" in vm_params, "Parameters already specific to the object must be preserved")
+        self.assertEqual(vm_params["name_vm1"], params["name_vm1"], "Parameters already specific to the object must be preserved")
+        self.assertFalse("name_vm2" in vm_params, "Parameters specific to a different object must be pruned")
+        self.assertFalse("name" in vm_params, "Parameters not specific to any object must be pruned if there is specific alternative")
+        self.assertTrue("surname_vm1" in vm_params, "Parameters not specific to any object must become specific to the object")
+        self.assertFalse("surname" in vm_params, "Parameters not specific to any object must become specific to the object")
```

Comment:
btw it's useful to list the expected and actual values (if not everything) instead just the message. Usually it's better to simply add a comment above the call, leave the default reporting and use the right methods like `assertIn`.

Reply:
Great suggestion, added.